src/main/java/SentinelRefactoring.java

```diff
@@ -0,0 +1,401 @@
+import java.util.Dictionary;
+import java.util.Hashtable;
+import java.util.List;
+
+import org.eclipse.jdt.core.dom.AST;
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.Assignment;
+import org.eclipse.jdt.core.dom.Block;
+import org.eclipse.jdt.core.dom.Expression;
+import org.eclipse.jdt.core.dom.ExpressionStatement;
+import org.eclipse.jdt.core.dom.IfStatement;
+import org.eclipse.jdt.core.dom.InfixExpression;
+import org.eclipse.jdt.core.dom.NullLiteral;
+import org.eclipse.jdt.core.dom.NumberLiteral;
+import org.eclipse.jdt.core.dom.PrefixExpression;
+import org.eclipse.jdt.core.dom.SimpleName;
+import org.eclipse.jdt.core.dom.rewrite.ASTRewrite;
+
+import org.eclipse.jdt.core.dom.Statement;
+import org.eclipse.jdt.core.dom.VariableDeclarationFragment;
+import org.eclipse.jdt.core.dom.VariableDeclarationStatement;
+
+/**
+ * This class represents a refactoring in which integer variables whose values
+ * represent the nullness of another variables are refactored into explicit null
+ * checks
+ */
+public class SentinelRefactoring extends Refactoring {
+	public static final String NAME = "SentinelRefactoring";
+	/**
+	 * List of all sentinels found during traversal of the AST
+	 */
+	private final Dictionary<String, Sentinel> sentinels;
+
+	/**
+	 * Used to skip reparsing AST elements that have already been analyzed
+	 */
+	private boolean skipSentinelDeclaration = false;
+
+	/**
+	 * Helper class for storing the values of a sentinel reference
+	 */
+	protected class Sentinel {
```

Comment:
This class doesn't have any actual behavior besides getters and setters, so it should be a `record` rather than a `class`.